test/test_helper/common.bash

```diff
@@ -194,7 +194,8 @@ function wait_until_change_detection_event_completes() {
   }
 
   function __is_changedetector_processing() {
-    [[ $(__change_event_status) == "${CHANGE_EVENT_START}" ]]
+    [[ $(__change_event_status) == "${CHANGE_EVENT_START}" ]] \
+    || [[ $(__change_event_status) == "${CHANGE_EVENT_END}" ]]
```

Comment:
@polarathene could you please double-check this for me? I think it's fine but not 100% sure whether the logic checks out - it should though, just making sure here.

Reply:
`CHANGE_EVENT_END` implies the `changedetector` service is no longer processing the event. This would risk causing false positives.

The correct approach to resolve the bug is described in [my comment on a related issue](https://github.com/docker-mailserver/docker-mailserver/issues/2966#issuecomment-1366973489).

I should have time to look into resolving this myself today or tomorrow.

---

I wasn't aware that `sleep` could use non-integer values for sub-second delays. That alone would probably be enough in most cases. If not adding in the counting mechanism should do the trick, possibly also adding a way to increase the polling window of the change detector service itself (for the purpose of tests at least).

Reply:
Sound's good. @polarathene it's not the polling interval of the changedetector that's interfering here, it's the time to process a change event. if it takes less time to process the event than the interval that checks it the changedetector has started processing the event, in the current code this causes the event status to be CHANGE_EVENT_END before it ever returned CHANGE_EVENT_START.

However i think the counting option would be a better solution for sure.